pkg/iac-providers/kubernetes/v1/extract-images.go

```diff
@@ -0,0 +1,152 @@
+package k8sv1
+
+import (
+	"encoding/json"
+	"github.com/accurics/terrascan/pkg/iac-providers/output"
+	"github.com/accurics/terrascan/pkg/utils"
+	yamltojson "github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	k8sappsv1 "k8s.io/api/apps/v1"
+	k8sbatchv1 "k8s.io/api/batch/v1"
+	k8sbatchv1beta1 "k8s.io/api/batch/v1beta1"
+	k8scorev1 "k8s.io/api/core/v1"
+)
+
+func (k *K8sV1) extractContainerImages(kind string, doc *utils.IacDocument) ([]output.ContainerNameAndImage, []output.ContainerNameAndImage, error) {
+	var containerImages = make([]output.ContainerNameAndImage, 0)
+	var initContainerImages = make([]output.ContainerNameAndImage, 0)
+	var data []byte
+	var err error
+
+	if doc.Data == nil {
+		return containerImages, initContainerImages, errors.Errorf("document does not have any resource data for unmarshalling")
+	}
+
+	if doc.Type == utils.YAMLDoc {
+		data, err = yamltojson.YAMLToJSON(doc.Data)
+		if err != nil {
+			return nil, nil, err
+		}
+	} else {
+		data = doc.Data
+	}
+
+	switch kind {
+	case "Pod":
+		pod := k8scorev1.Pod{}
+		err = json.Unmarshal(data, &pod)
+		if err != nil {
+			err := errors.Errorf("error unmarshalling pod: %v", err)
+			zap.S().Errorf(err.Error())
+			return nil, nil, err
+		}
+		for _, container := range pod.Spec.Containers {
+			containerImages = append(containerImages, output.ContainerNameAndImage{Name: container.Name, Image: container.Image})
+		}
+		for _, container := range pod.Spec.InitContainers {
+			initContainerImages = append(initContainerImages, output.ContainerNameAndImage{Name: container.Name, Image: container.Image})
```

Comment:
This code block is repetitive for all cases, can we create a separate function as below?

```
func readContainers(containers []k8scorev1.Container) []output.ContainerNameAndImage {
	containerImages := make([]output.ContainerNameAndImage, 0)
	for _, container := range containers {
		containerImages = append(containerImages, output.ContainerNameAndImage{Name: container.Name, Image: container.Image})
	}
	return containerImages
}
```

```
containerImages = append(containerImages, readContainers(pod.Spec.Containers)...)
initContainerImages = append(initContainerImages, readContainers(pod.Spec.InitContainers)...)
```